src/Microsoft.Azure.SignalR/EndpointUtility/ServiceEndpointUtility.cs

```diff
@@ -145,11 +145,22 @@ internal static (string, string) ParseConnectionString(string connectionString)
 
                 if (dict.ContainsKey(EndpointProperty) && dict.ContainsKey(AccessKeyProperty))
                 {
+                    if (!ValidateEndpoint(dict[EndpointProperty]))
+                    {
+                        throw new ArgumentException($"Endpoint is invalid: {dict[EndpointProperty]}");
```

Comment:
-> `Endpoint property in connection string is not a valid URI: {dict[EndpointProperty]}`.